Description: adding option to define root where to load npm modules
Author: Maurits Lamers <maurits@weidestraat.nl>
Forwarded: not-needed
Reviewed-By Xavier Guimard <yadd@debian.org>
Last-Update: 2020-02-21

--- a/lib/grunt/task.js
+++ b/lib/grunt/task.js
@@ -7,6 +7,7 @@
 
 // Nodejs libs.
 var path = require('path');
+var fs = require('fs');
 
 // Extend generic "task" util lib.
 var parent = grunt.util.task.create();
@@ -367,9 +368,10 @@
 
 // Load tasks and handlers from a given locally-installed Npm module (installed
 // relative to the base dir).
-task.loadNpmTasks = function(name) {
+task.loadNpmTasks = function(name, root) {
   loadTasksMessage('"' + name + '" local Npm module');
-  var root = path.resolve('node_modules');
+  root =  [ root, 'node_modules', '/usr/share/nodejs', '/usr/lib/nodejs'].find( v => v && fs.existsSync( path.join(v, name) ) );
+  root = root || '/usr/share/nodejs';
   var pkgpath = path.join(root, name);
   var pkgfile = path.join(pkgpath, 'package.json');
   // If package does not exist where grunt expects it to be,
